### DISCUSSION TITLE
Seagate head parking - need assistance / guidance

### DISCUSSION THREAD

**Discussion by talormanda** [Seagate/openSeaChest#82] [Q&A] [answered]
I was given an almost new Seagate BarraCuda Pro 4TB (ST4000DM006). It works great except for the fact that my sensitive hearing always tunes into what I believe to be the head parking every 9-10 seconds. I am not an expert with drives, but is there something I can do to prevent this from happening so much? Googling tells me perhaps this the EPC setting that I can change with openSeaChest or SeaChestUtilities.

I am looking for clarification / reassurance if anything. If changing the setting is bad, what to set it to, etc. I am unsure what the Idle a,b,c,standby_z things are in the below command.

```
C:\Users\talor\Downloads\SeaChestUtilities\Windows\Win64-Non-RAID>SeaChest_PowerControl_x64_windows.exe -d PD0 --showEPCSettings
==========================================================================================
 SeaChest_PowerControl - Seagate drive utilities - NVMe Enabled
 Copyright (c) 2014-2021 Seagate Technology LLC and/or its Affiliates, All Rights Reserved
 SeaChest_PowerControl Version: 3.0.2-2_2_3 X86_64
 Build Date: Jun 17 2021
 Today: Wed Apr  6 10:29:58 2022        User: talor (admin)
==========================================================================================

\\.\PhysicalDrive0 - ST4000DM006-2G5107 - ZC1CJJHH - ATA
.

===EPC Settings===
        * = timer is enabled
        C column = Changeable
        S column = Savable
        All times are in 100 milliseconds

Name       Current Timer Default Timer Saved Timer   Recovery Time C S
Idle A     *1            *1            *1            1             Y Y
Idle B     *1200         *1200         *1200         4             Y Y
Idle C      0             6000          6000         42            Y Y
Standby Z   0             9000          9000         190           Y Y
```

**Comment by vonericsen** [comment, marked as answer]
Hi @talormanda,

Great question!
The EPC feature (Extended Power Conditions) is designed for power savings, and how a vendor accomplishes this may vary.
The idle states and standby states go from most power used to least power used in this order according to the standard:
>idle_a -> idle_b -> idle_c -> standby_y -> standby_z

Seagate markets this feature as "Power Choice" and has a document describing how it works and the energy savings it provides and what happens in the different modes [here](https://www.seagate.com/files/docs/pdf/en-GB/whitepaper/tp608-powerchoice-tech-provides-gb.pdf).

In the document above, it describes the heads are parking in idle_b and lower states, so this is the transition you are likely hearing from your drive when it goes into this lower power mode as it is not receiving commands for that time period and starts saving power by parking the heads.
The timer value we see shows this happens after 2 minutes of inactivity (1200 * 100 = 120000 milliseconds).
After the head has parked, when the drive receives a command requiring media access (read, write and some others) it will load the head over the media again to service these commands.

You have a couple options on how to solve this.
1. You can disable the idle_b timer with `openSeaChest_PowerControl -d <handle> --idle_b disable`
2. You can adjust the timer for the idle_b so that it goes into this state less frequently `openSeaChest_PowerControl -d <handle> --idle_b newTimerValue`. 

Even if you change this timer or disable it, the operating system may request parking the heads with the idle command specifying an unload. I do not know if this will be issued by your OS though. 

There will be no harm disabling idle_b as long as you do not care about the power savings it gives. I do not have a good recommendation for the timer value right now and you may find that it does not help as it would change the frequency you hear this at instead.
Since the other states, idle_c and standby_z, are not active, you do not need to worry about them right now, but could use a similar process to change them with these tools.

I hope this helps. Let me know if you have any other questions.

**Reply by talormanda** [reply]
Yes this does help. I am testing it now and I believe it is working. I set it to 30 minutes. You are correct that the OS (I am on windows 11) could also be causing the head to park because when I was hearing the chirping, it was every 9 seconds which isn't the 2 minutes it is set to. But once the OS was on for a bit..this seemed to stop. So I suspect the OS is doing some indexing or disk verification on boot for a bit. I will reply back if my testing changes but so far so good.

**Comment by GeoffreyAA** [comment]
Hi @vonericsen,

Apologies for replying to an old thread.

I tried disabling `idle_b` on an external, USB-connected Seagate drive (ST2000LM007), and it failed, saying, "Configuring EPC Settings is not supported on this device."

Is changing the head-parking settings not possible on USB-connected drives?

Thanks.

**Reply by vonericsen** [reply]
Hi @GeoffreyAA,

Being connected over USB does not necessarily mean you cannot change EPC settings.
What is more likely is that the drive firmware does not allow changing the EPC settings. That can be because EPC itself is newer than the drive, or the feature was not enabled in the firmware, or something like that.

From the MN, this drive may not support EPC. You would need to look at the `-i --SATInfo` output to know for sure.

Prior to EPC you can change the APM settings (volatile, needs reprogramming after each power cycle) or the older standby timer (volatile, needs reprogramming after each power cycle).

If you set an APM value above I think 80h (127) then it will not go into standby mode (stay spinning). If you set the value to 255 then that is the highest power use and should prevent parking heads.
If you set the legacy standby timer value (--standby timevaluehere) with openSeaChest_PowerControl, then this controls the time before the drive parks the heads and spins down.
With APM and this older standby timer you do need to understand that the drive will choose whichever timer expires soonest to control spinning down. If APM is a low enough value to trigger a standby change before the legacy timer value, then that will spin it down. If APM is high and the legacy standby timer is programmed to a lower value, then the legacy timer tells the firmware when to spindown.

There were no clear rules prior to EPC on when the drive firmware was to do head parking (other than when spinning the whole drive down) so there is a lot of trial and error with these older methods to figure it out. APM is just a range of values that a vendor can use to scale the internal firmware algorithm for power management with the only real rules being 255 = highest power, 127 = intermediate, 0 = lowest with standby enabled. A vendor is not required to implement a different timer change for the entire range. They could choose to implement only 3 if they wanted to, but to the best of my knowledge there is some scaling in the drive firmware of Seagate drives that changes the timers by some amount of time (I just don't know exactly what that is since I don't have access to the drive firmware itself).

**Reply by GeoffreyAA** [reply]
Thank you for the detailed, highly helpful response, @vonericsen. I appreciate it.

Since this is a new drive, manufactured in Oct. 2025, I expect it is the firmware that doesn't allow changing the EPC settings. As soon as I get the opportunity, I will try the `-i --SATInfo` command and report back.
